dynamic_db_router/router.py

```diff
@@ -83,45 +89,33 @@ def lowest_id_account():
     def __init__(self, database, read=True, write=False):
         self.read = read
         self.write = write
+        self.database = database
         self.created_db_config = False
-        if isinstance(database, str):
-            self.database = database
-        elif isinstance(database, dict):
-            # Note: this invalidates the docs above. Update them
-            # eventually.
+        if isinstance(database, dict):
             self.created_db_config = True
             self.unique_db_id = str(uuid4())
             connections.databases[self.unique_db_id] = database
             self.database = self.unique_db_id
-        else:
-            msg = ("database must be an identifier for an existing db, "
-                   "or a complete configuration.")
-            raise ValueError(msg)
-
+ 
     def __enter__(self):
-        if not hasattr(THREAD_LOCAL, 'DB_FOR_READ_OVERRIDE'):
-            THREAD_LOCAL.DB_FOR_READ_OVERRIDE = ['default']
-        if not hasattr(THREAD_LOCAL, 'DB_FOR_WRITE_OVERRIDE'):
-            THREAD_LOCAL.DB_FOR_WRITE_OVERRIDE = ['default']
-        read_db = (self.database if self.read
-                   else THREAD_LOCAL.DB_FOR_READ_OVERRIDE[-1])
-        write_db = (self.database if self.write
-                    else THREAD_LOCAL.DB_FOR_WRITE_OVERRIDE[-1])
-        THREAD_LOCAL.DB_FOR_READ_OVERRIDE.append(read_db)
-        THREAD_LOCAL.DB_FOR_WRITE_OVERRIDE.append(write_db)
+        self.original_read_db = DB_FOR_READ_OVERRIDE.get()
```

Comment:
When are the __enter__ and __exit__ methods called in this context?

Reply:
not sure if i understand this question... Isn't the enter and exit method called automatically when using with? enter should be called at the beginning and exit at the end: [doc](https://docs.python.org/3/library/stdtypes.html#context-manager-types)

Reply:
I should have noted this is for my own education ;) I'd figured but I'd never written a context manager for Python so I wanted to confirm. Thanks for the learning, David!